### PR TITLE
feat: Add config based sec-group parsing and implementation

### PR DESCRIPTION
**Pull request by jwitko**
The only reason this is marked a draft is because I havent implemented individual modules per region we currently support but I wanted to run it up the flag pole before doing that to make it easier to review.

### CODE REVIEW

File: terraform/locals.aws.tf [hunk 1]
```diff
@@ -49,6 +49,98 @@ locals {
...
+          ingress_sg_name = "relay"
+        },
+        {
+          name        = "bp-egress-anywhere"
+          egress      = true
+          rule        = "all-all"
+          description = "all traffic egress to anywhere"
+        }
+      ]
+    },
```
Comment (wolf31o2) [inline terraform/locals.aws.tf:81]: Get rid of the `core` group.

File: terraform/locals.aws.tf [hunk 2]
```diff
@@ -49,6 +49,98 @@ locals {
...
+    {
+      name        = "relay"
```
Comment (wolf31o2) [inline terraform/locals.aws.tf:83]: Rename this to simply "node" or "cardano-node"

File: terraform/locals.aws.tf [hunk 3]
```diff
@@ -49,6 +49,98 @@ locals {
...
+      description = "Security group for relay nodes"
+      security_groups = [
```
Comment (wolf31o2) [inline terraform/locals.aws.tf:85]: Maybe something like "rules" since these aren't separate groups.

**Review by wolf31o2 — CHANGES_REQUESTED** [review]
Change "relay" to just "node" throughout, since they're the same thing.

File: terraform/locals.aws.tf (hunk 2 above)
Comment (jwitko) [reply terraform/locals.aws.tf:83]: Changed

File: terraform/locals.aws.tf (hunk 3 above)
Comment (jwitko) [reply terraform/locals.aws.tf:85]: Changed

File: terraform/locals.aws.tf (hunk 1 above)
Comment (jwitko) [reply terraform/locals.aws.tf]: Removed

**Comment by jwitko** [comment]
> Change "relay" to just "node" throughout, since they're the same thing.

Should be good to go on this

**Review by wolf31o2 — APPROVED** [review]

**Pull request merged**